sstables/mp_row_consumer.hh

```diff
@@ -509,6 +509,11 @@ public:
     //}
     proceed do_consume_cell(bytes_view col_name, int64_t timestamp, int64_t ttl, int64_t expiration, CreateCell&& create_cell) {
         struct column col(*_schema, col_name, timestamp);
+        auto cell_str = sstring(to_sstring_view(col.cell));
+        auto dc = _schema->dropped_columns().find(cell_str);
+        if (!col.cdef && col.cell.size() > 0 && (dc == _schema->dropped_columns().end() || dc->second.timestamp < timestamp)) {
```

Comment:
How about evaluating dc only `if (!col.cdef && col.cell.size() > 0)`?

Reply:
I optimised the condition and pulled the code to a single helper function.